### PR TITLE
GCC 8.3

### PR DESCRIPTION
**Pull request by DesWurstes**
I haven't built it.

### CODE REVIEW

File: Formula/gcc.rb [hunk 1]
```diff
@@ -4,17 +4,13 @@ class Gcc < Formula
   head "https://gcc.gnu.org/git/gcc.git"
 
   stable do
```
Comment (fxcoudert) [inline Formula/gcc.rb]: Please remove the `stable do` block, and move the `url`, `mirror` and `sha256` fields above `head`.

File: Formula/gcc.rb [hunk 2]
```diff
@@ -4,17 +4,13 @@ class Gcc < Formula
...
-    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724
-    patch :DATA
+    url "https://ftp.gnu.org/gnu/gcc/gcc-8.3.0/gcc-8.3.0.tar.xz"
+    mirror "https://ftpmirror.gnu.org/gcc/gcc-8.3.0/gcc-8.3.0.tar.xz"
+    sha256 "64baadfe6cc0f4947a84cb12d7f0dfaf45bb58b7e92461639596c21e02d97d2c"
   end
 
   bottle do
-    rebuild 2
+    rebuild 1
```
Comment (fxcoudert) [inline Formula/gcc.rb]: Do not touch this.

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Changes should be squashed into a single commit, with message `gcc 8.3.0`

**Review by fxcoudert — APPROVED** [review]

**Review by zbeekman — DISMISSED** [review]
Failing tests. `nwchem` uses ancient style Fortran, maybe gcc 8.3 legitimately broke it, or maybe it just needs a revision bump. The python failures on high-sierra are suspect. Not sure what is going on there. We should resolve the failures before we ship this.

**Comment by zbeekman** [comment]
I gave `libbi` and `nwchem` a revision bump to see if that resolves the issue.

**Comment by fxcoudert** [comment]
`libbi` test is preexisting, sadly. I've reported it upstream: https://github.com/lawmurray/LibBi/issues/45

We've already rev-bumped that formula twice in 1 month: https://github.com/Homebrew/homebrew-core/pull/36335 and https://github.com/Homebrew/homebrew-core/pull/36432 to get rid of this test failure. We probably need to understand what is happening.

**Comment by fxcoudert** [comment]
Let's run the `libbi` test without OpenMP, which succeeds:

```
diff --git a/Formula/libbi.rb b/Formula/libbi.rb
index fd44f89ae1..f85388738a 100644
--- a/Formula/libbi.rb
+++ b/Formula/libbi.rb
@@ -137,7 +137,7 @@ class Libbi < Formula
 
   test do
     cp Dir[pkgshare/"Test.bi", pkgshare/"test.conf"], testpath
-    system "#{bin}/libbi", "sample", "@test.conf"
+    system "#{bin}/libbi", "sample", "@test.conf", "--disable-openmp"
     assert_predicate testpath/"test.nc", :exist?
   end
 end
```

**Comment by fxcoudert** [comment]
Thanks @DesWurstes

**Comment by JayBrown** [comment]
Failed to build on Mojave:

```
Last 15 lines from ~/Library/Logs/Homebrew/gcc/02.make:
cc1plus: warning: -Wabi won't warn about anything [-Wabi]
cc1plus: note: -Wabi warns about differences from the most up-to-date ABI, which is also used by default
cc1plus: note: use e.g. -Wabi=11 to warn about changes from GCC 7
../../../../../libstdc++-v3/src/filesystem/path.cc:29:10: fatal error: experimental/filesystem: No such file or directory
 #include <experimental/filesystem>
          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[6]: *** [path.lo] Error 1
make[6]: *** Waiting for unfinished jobs....
make[5]: *** [all-recursive] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all] Error 2
make[2]: *** [all-stage2-target-libstdc++-v3] Error 2
make[1]: *** [stage2-bubble] Error 2
make: *** [all] Error 2
```

Doctor output: `Warning: Skipping gcc: most recent version 8.3.0 not installed
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!`

**Comment by fxcoudert** [comment]
@JayBrown does installing the bottle work for you?

**Comment by JayBrown** [comment]
Just tested it with `brew upgrade --force-bottle gcc`, and it works. One of the reasons that brew wanted to build it from source was seemingly that it was unable to detect the Command Line Tools. I have installed Xcode, the CLT are there too, but brew doesn't seem to notice. (?)

**Comment by zbeekman** [comment]
Not sure it this applies to you, but after upgrading to Mojave, I believe you need to (re)install CLT/XCode.

**Pull request closed without merging**